tests/jballerina-unit-test/src/test/resources/test-src/typedefs/type-definitions.bal

```diff
@@ -338,6 +338,29 @@ public function returnTupleWithSingletonType2() returns X2 {
     return ["x", 2, ()];
 }
 
+type Data record {|
+    string value;
+|};
+
+const annotation Data dataAnon on type;
+
+@dataAnon {
+    value: "T1"
+}
+type Person2 record {|
+    string name;
+|};
+
+function testAnnotWithRecordTypeDefinition() {
+    Person2 foo = {name: "James"};
+    typedesc<any> t = typeof foo;
+    Data? annon = t.@dataAnon;
+    assertEquality("{\"value\":\"T1\"}", annon.toString());
```

Comment:
Generally better to assert the value rather than the string representation IMO (since it'll take types also into account).